src/cluster/clusterServices.tsx

```diff
@@ -0,0 +1,360 @@
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': API_HEADER_CONTENT_TYPE,
+            Authorization: API_HEADER_BEARER + credentials.access_token
+          }
+        }
+      )
+        .then((response: Response) => {
+          response
+            .json()
+            .then((responseResult: IClusterDetailsResponse) => {
+              if (responseResult.error && responseResult.error.code === 404) {
+                setErrorView(true);
+              }
+              if (responseResult?.error?.code) {
+                toast.error(
+                  responseResult?.error?.message,
+                  toastifyCustomStyle
+                );
+              }
+              setClusterInfo(responseResult);
+              setIsLoading(false);
+            })
+            .catch((e: Error) => {
+              console.log(e);
+              setIsLoading(false);
+            });
+        })
+        .catch((err: Error) => {
+          setIsLoading(false);
+          console.error('Error listing clusters Details', err);
+          DataprocLoggingService.log(
+            'Error listing clusters Details',
+            LOG_LEVEL.ERROR
+          );
+          toast.error(
+            `Failed to fetch cluster details ${clusterSelected}`,
+            toastifyCustomStyle
+          );
+        });
+    }
+  };
+
+  static statusApiService = async (
+    selectedCluster: string,
+    listClustersAPI: () => void,
+    timer: any
+  ) => {
+    try {
+      const response = await authenticatedFetch({
+        uri: `clusters/${selectedCluster}`,
+        method: HTTP_METHOD.GET,
+        regionIdentifier: 'regions'
+      });
+      const formattedResponse = await response.json();
+
+      if (formattedResponse.status.state === ClusterStatus.STATUS_STOPPED) {
+        ClusterService.startClusterApi(selectedCluster);
+        clearInterval(timer.current);
+      }
+      if (formattedResponse?.error?.code) {
+        toast.error(formattedResponse?.error?.message, toastifyCustomStyle);
+      }
+      listClustersAPI();
+    } catch (error) {
+      console.error('Error fetching status', error);
+      DataprocLoggingService.log('Error fetching status', LOG_LEVEL.ERROR);
+      toast.error(
+        `Failed to fetch the status ${selectedCluster}`,
+        toastifyCustomStyle
+      );
+    }
+  };
+
+  static restartClusterApiService = async (
+    selectedCluster: string,
+    setRestartEnabled: (value: boolean) => void,
+    listClustersAPI: () => void,
+    timer: any,
+    statusApi: (value: string) => void
+  ) => {
+    setRestartEnabled(true);
+
+    try {
+      const response = await authenticatedFetch({
+        uri: `clusters/${selectedCluster}:stop`,
+        method: HTTP_METHOD.POST,
+        regionIdentifier: 'regions'
+      });
+      const formattedResponse = await response.json();
+      console.log(formattedResponse);
+      listClustersAPI();
+      timer.current = setInterval(() => {
```

Comment:
when does this timer get cleared?

Reply:
Cleared in statusApiService After cluster status changed to 'STOP' clearInterval executed